File: app/graphql/object_types/task.rb
# frozen_string_literal: true

module ObjectTypes
  class Task < Types::BaseObject
    field :id, ID, null: false
    field :title, String, null: false
    field :body, String
    field :state, Enum::TaskStateEnum, null: false
    field :limit_on, GraphQL::Types::ISO8601Date
    field :user_id, Integer, null: false
    field :created_at, GraphQL::Types::ISO8601DateTime, null: false
    field :updated_at, GraphQL::Types::ISO8601DateTime, null: false
    field :image_url, String

    field :labels, [ObjectTypes::Label], null: false

    def labels
      Loaders::AssociationLoader.for(::Task, :labels).load(object)
    end
  end
end

File: app/javascript/graphql/queries/task.ts
import { gql } from "@apollo/client";

export default gql`
  query task($id: ID!) {
    task(id: $id) {
      id
      title
      body
<<<<<<< HEAD
      state
      limitAt
=======
      limitOn
>>>>>>> b3bc1ab7ff040b136dc46a7cb41caddaaeedc5e1
    }
  }
`;
